src/main/java/nextstep/subway/path/domain/PathFinder.java

```diff
@@ -0,0 +1,56 @@
+package nextstep.subway.path.domain;
+
+import nextstep.subway.line.domain.Distance;
+import nextstep.subway.line.domain.Section;
+import nextstep.subway.path.dto.PathStation;
+import org.jgrapht.GraphPath;
+import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
+import org.jgrapht.graph.DefaultWeightedEdge;
+import org.jgrapht.graph.WeightedMultigraph;
+
+import java.util.List;
+import java.util.Optional;
+
+public class PathFinder {
+    private static WeightedMultigraph<PathStation, DefaultWeightedEdge> graph;
+
+    public PathFinder(List<Section> sections) {
+        graph = new WeightedMultigraph(DefaultWeightedEdge.class);
+        addAllSections(sections);
+    }
+
+    public List<PathStation> findPath(PathStation source, PathStation target) {
+        return createGraphPath(source, target).getVertexList();
+    }
+
+    public Distance findShortestDistance(PathStation source, PathStation target) {
```

Comment:
```suggestion
    public Distance findShortestDistance(PathStation source, PathStation target) {
        double weight = createGraphPath(source, target).getWeight();
        return new Distance((int) weight);
    }
```
다른 개발자도 읽기 좋은 코드를 작성해주세요
한줄에 다 쓰면 유지보수 하기 힘들어집니다

Reply:
확실히 더 가독성이 좋아졌군요!
무조껀 숏코딩을 한다고 좋은게 아닌데, 이후엔 가독성을 고려하여 작성하겠습니다👍